common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -883,7 +883,16 @@ public void channelReadComplete(ChannelHandlerContext ctx) {
 
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
-        log.error("[{}] Unexpected Exception", sessionId, cause);
+        if (cause instanceof IOException) {
+            if (log.isInfoEnabled()) {
+                log.info("[{}][{}] IOException: {}", sessionId,
```

Comment:
No stack trace here. I know it is not needed in 99% of the cases, but I suggest logging it if debug or trace level is enabled.

Reply:
Here is no stack trace. Only getMessage() appeared . Now it logs stacktrace every time. IOException is not so rare. Any network disconnect fires IOException and in practice - that is a lot 99,99% of all error messages. This provide you the information that the device is disconnected. I agree that debug level is even better than the info.